test/src/helper/spawn.ts

```diff
@@ -56,7 +56,7 @@ export type ChainType =
 type ProcessState =
     | { state: "stopped" }
     | { state: "initializing" }
-    | { state: "running"; process: ChildProcess }
+    | { state: "running"; process: ChildProcess; readline: ReadLine }
```

Comment:
Wouldn't it be better to attach `readline` to `CodeChain` rather than `ProcessState`?
It always exists when the instance is alive.

Reply:
ReadLine's lifetime follows the child process's lifetime. The only piping stream is from the process, so it is gone as soon as the process is gone.